front/lib/connector_providers.ts

```diff
@@ -171,3 +173,82 @@ export function getConnectorProviderLogoWithFallback(
   }
   return CONNECTOR_CONFIGURATIONS[provider].logoComponent;
 }
+
+export const isValidConnectorSuffix = (suffix: string): boolean => {
+  return /^[a-z0-9\-_]{1,16}$/.test(suffix);
+};
+
+export const isConnectorProviderAllowedForPlan = (
```

Comment:
I guess this code is duplicated from the existing logic, right? If so, how easy would it be to use this new code in the old one?